src/Kfz24/Client/Aws/SqsClient.php

```diff
@@ -57,34 +22,84 @@ class SqsClient extends AbstractAwsClient
      */
     private $logger;
 
+    /**
+     * @var LargePayloadMessageExtension
+     */
+    private $largePayloadMessageExtension;
+
     /**
      * @param MessageValidator $validator
+     *
+     * @return SqsClient
      */
-    public function setValidator(MessageValidator $validator)
+    public function setValidator(MessageValidator $validator): SqsClient
     {
         $this->validator = $validator;
+
+        return $this;
     }
 
     /**
      * @param LoggerInterface $logger
+     *
+     * @return SqsClient
      */
-    public function setLogger(LoggerInterface $logger)
+    public function setLogger(LoggerInterface $logger): SqsClient
     {
         $this->logger = $logger;
+
+        return $this;
+    }
+
+    /**
+     * @param LargePayloadMessageExtension $largePayloadMessageExtension
+     *
+     * @return SqsClient
+     */
+    public function setLargePayloadMessageExtension(
+        LargePayloadMessageExtension $largePayloadMessageExtension
+    ): SqsClient {
+        $this->largePayloadMessageExtension = $largePayloadMessageExtension;
+
+        return $this;
     }
 
     /**
      * @param string|array $message
      *
      * @return Result
      *
+     * @throws \Exception
      */
     public function send($message)
     {
-        if (!is_array($message) || !array_key_exists('MessageBody', $message)) {
-            $message = ['MessageBody' => json_encode($message)];
+        if (is_array($message)) {
```

Comment:
Isn't there a better way to write these statements? This feel overcomplicated to me.

Reply:
Yes it feels complicated but to catch all these use-cases we need these checks. I could pull them out into 2 methods `prepareMessageFromArrayMessage()` and `prepareMessageFromNonArrayMessage()` but the conditions will still be there then.

Reply:
Yes, maybe it's a good idea to move them into separate methods.

Reply:
Yes, maybe it's a good idea to move them into separate methods.